torch/csrc/autograd/VariableTypeUtils.h

```diff
@@ -178,19 +177,20 @@ inline Tensor as_view(const Tensor & base, const Tensor & tensor, bool is_bw_dif
   if (is_fw_differentiable) {
     // Check if base is a forward differentiable view
     auto base_meta = torch::autograd::impl::get_autograd_meta(base);
-    auto is_view = base_meta && base_meta->is_view_;
-    if (is_view && static_cast<DifferentiableViewMeta*>(base_meta)->has_fw_view()) {
-      auto diff_view_meta = static_cast<DifferentiableViewMeta*>(base_meta);
-      const auto& base_fw_info = diff_view_meta->get_forward_view();
-      new_fw_info = base_fw_info.chain(base, tensor, view_func);
+    if (base_meta && base_meta->is_view_) {
+      auto diff_view_meta = torch::autograd::impl::get_view_autograd_meta(base);
+      if (diff_view_meta->has_fw_view()) {
```

Comment:
Is it intended that the behavior here changed? If this if condition is false, then we will no longer go into the else.

Reply:
Good catch.
I wanted to reduce the amount of time we access the autograd meta but that's not really possible I guess.
I'll revert back to the old logic